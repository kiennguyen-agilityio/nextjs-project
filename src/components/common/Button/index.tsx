import type { ReactNode } from 'react';

type ButtonProps = {
  children: ReactNode;
  startIcon?: ReactNode;
  endIcon?: ReactNode;
  type?: 'button' | 'submit';
  variant?:
    | 'outline'
    | 'primary'
    | 'secondary'
    | 'success'
    | 'error'
    | 'info'
    | 'warning';
  customClass?: string;
  name?: string;
  value?: string;
  ariaLabel?: string;
  disabled?: boolean;
  onClick?: () => void;
};

export const Button = ({
  children,
  startIcon,
  endIcon,
  type = 'button',
  variant = 'primary',
  customClass,
  name,
  value,
  ariaLabel,
  disabled,
  onClick,
}: ButtonProps) => {
  const baseClass = 'flex items-center rounded p-2';
  let stateClass = '';

  switch (variant) {
    case 'primary':
      stateClass = `bg-[#4270ec] text-white hover:bg-blue-600 ${disabled ? 'hover:cursor-not-allowed hover:opacity-70' : `hover:opacity-80`}`;
      break;
    case 'secondary':
      stateClass = `bg-white text-[#62656e] hover:bg-gray-300 ${disabled ? 'hover:cursor-not-allowed hover:opacity-70' : `hover:opacity-80`}`;
      break;
    case 'success':
      stateClass = `bg-green-600 text-white hover:bg-green-700 ${disabled ? 'hover:cursor-not-allowed hover:opacity-70' : `hover:opacity-80`}`;
      break;
    case 'warning':
      stateClass = `bg-amber-400 text-black hover:bg-amber-500 ${disabled ? 'hover:cursor-not-allowed hover:opacity-70' : `hover:opacity-80`}`;
      break;
    case 'error':
      stateClass = `bg-rose-500 text-white hover:bg-rose-600 ${disabled ? 'hover:cursor-not-allowed hover:opacity-70' : `hover:opacity-80`}`;
      break;
    case 'outline':
      stateClass = `border border-gray-200 dark:border-gray-700 hover:bg-gray-100 ${disabled ? 'hover:cursor-not-allowed hover:opacity-70' : `hover:opacity-80`}`;
      break;

    default:
      break;
  }

  return (
    <button
      type={type}
      name={name}
      value={value}
      aria-label={ariaLabel}
      className={[baseClass, stateClass, customClass].join(' ')}
      disabled={disabled}
      onClick={onClick}
    >
      {startIcon && <span className="mr-2">{startIcon}</span>}
      {children}
      {endIcon && <span className="ml-r">{endIcon}</span>}
    </button>
  );
};
