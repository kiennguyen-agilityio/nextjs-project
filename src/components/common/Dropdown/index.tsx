interface DropdownProps {
  label: string | number;
  options: Array<{ value: string; label: string }>;
  customClass?: string;
}

const Dropdown = ({ label, options, customClass }: DropdownProps) => (
  <select
    className={`inline-flex justify-center w-full rounded-3xl px-4 py-2 text-sm font-medium text-gray-700 hover:bg-gray-200 focus:outline-none max-w-32 ${customClass}`}
  >
    <option value="">{label}</option>
    {options.map((option) => (
      <option key={option.value} value={option.value}>
        {option.label}
      </option>
    ))}
  </select>
);

export default Dropdown;
