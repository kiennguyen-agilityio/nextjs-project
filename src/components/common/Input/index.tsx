import { HTMLInputTypeAttribute } from 'react';

interface Props {
  name: string;
  label: string;
  type: HTMLInputTypeAttribute;
  placeholder?: string;
}

export const Input = ({ name, label, type, placeholder }: Props) => (
  <div>
    <label
      className="block text-dark-100 text-sm font-medium leading-5 mb-4"
      htmlFor={name}
    >
      {label}
    </label>
    <input
      className="rounded-[10px] w-full h-12 py-2 px-4 bg-white-300 text-dark-90 placeholder-dark-50 leading-tight border border-gray-300 focus:outline-none focus:shadow-outline focus:ring-2 focus:ring-slate-300"
      id={name}
      name={name}
      type={type}
      placeholder={placeholder}
    />
  </div>
);
