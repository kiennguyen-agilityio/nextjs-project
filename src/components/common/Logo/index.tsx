export const Logo = () => {
  return (
    <div className="flex flex-row items-center leading-none text-white">
      <p className="text-4xl">Admin Dashboard</p>
    </div>
  );
};
