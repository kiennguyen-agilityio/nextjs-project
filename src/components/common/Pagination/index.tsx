'use client';
import { usePathname, useSearchParams } from 'next/navigation';

// components
import PaginationArrow from '@/components/common/Pagination/PaginationArrow';

// constants
import { DEFAULT_LIMIT } from '@/constants/defaultValue';

interface PaginationProps {
  totalUsers: number;
  limit: number;
  currentPage: number;
}

export default function Pagination({
  totalUsers,
  limit,
  currentPage,
}: PaginationProps) {
  const pathname = usePathname();
  const searchParams = useSearchParams();
  const startItem = (currentPage - 1) * limit + 1;
  const endItem = Math.min(currentPage * limit, totalUsers);
  const totalPages = Math.ceil(totalUsers / limit);

  const createPageURL = (pageNumber: number | string) => {
    const params = new URLSearchParams(searchParams);
    params.set('page', pageNumber.toString());
    params.set('limit', limit.toString());
    return `${pathname}?${params.toString()}`;
  };

  const itemsPerPage = DEFAULT_LIMIT;

  return (
    <div className="flex items-center justify-between">
      <div className="text-sm">
        Displaying <strong>{itemsPerPage}</strong> out of {totalPages}
      </div>

      <div className="flex items-center">
        <div className="flex items-center mx-2">
          <span className="text-sm font-medium">{startItem}</span>
          <span className="mx-1">-</span>
          <span className="text-sm font-medium">{endItem}</span>
        </div>
        <PaginationArrow
          direction="left"
          href={createPageURL(currentPage - 1)}
          isDisabled={currentPage <= 1}
        />

        <PaginationArrow
          direction="right"
          href={createPageURL(currentPage + 1)}
          isDisabled={currentPage >= totalPages}
        />
      </div>
    </div>
  );
}
