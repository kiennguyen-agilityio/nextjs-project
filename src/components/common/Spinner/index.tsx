const Spinner = () => {
  return (
    <div
      role="status"
      className="w-4 h-4 border-4 border-t-transparent border-gray-500 rounded-full animate-spin"
    />
  );
};

export default Spinner;
