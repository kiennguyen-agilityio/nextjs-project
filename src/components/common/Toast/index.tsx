import React, { useEffect } from 'react';

interface ToastProps {
  message: string;
  type: 'success' | 'error' | '';
  onClose: () => void;
}

const Toast: React.FC<ToastProps> = ({ message, type, onClose }) => {
  useEffect(() => {
    const timer = setTimeout(() => {
      onClose();
    }, 3000);
    return () => clearTimeout(timer);
  }, [onClose]);

  return (
    <div
      className={`fixed top-4 right-4 px-4 py-2 rounded shadow-lg ${type === 'success' ? 'bg-green-500' : 'bg-red-500'} text-white`}
    >
      {message}
    </div>
  );
};

export default Toast;
