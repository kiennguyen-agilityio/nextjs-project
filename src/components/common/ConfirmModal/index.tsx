'use client';
import { useState } from 'react';

// components
import Button from '@/components/common/Button';

interface ConfirmationModalProps {
  isOpen: boolean;
  onConfirm: () => void;
  onCancel: () => void;
  message: string;
}

const ConfirmationModal = ({
  isOpen,
  onConfirm,
  onCancel,
  message,
}: ConfirmationModalProps) => {
  const [loading, setLoading] = useState(false);

  const handleConfirm = async () => {
    setLoading(true);
    await onConfirm();
    setLoading(false);
  };

  if (!isOpen) return null;

  return (
    <div className="fixed inset-0 flex items-center justify-center bg-gray-500 bg-opacity-50">
      <div className="bg-white p-6 rounded-lg shadow-lg">
        <p className="mb-4">{message}</p>
        <div className="flex justify-end">
          <Button variant="secondary" onClick={onCancel} customClass="mr-2">
            Cancel
          </Button>
          <Button variant="error" onClick={handleConfirm} loading={loading}>
            Confirm
          </Button>
        </div>
      </div>
    </div>
  );
};

export default ConfirmationModal;
