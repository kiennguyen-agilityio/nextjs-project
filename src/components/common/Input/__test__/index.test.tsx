import { render, screen } from '@testing-library/react';

// component
import { Input } from '@/components/common/Input';

const name = 'testName';
const label = 'Test Label';
const type = 'text';
const placeholder = 'Test Placeholder';

describe('Input', () => {
  const mockOnChange = jest.fn();

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <Input
        name={name}
        label={label}
        type={type}
        placeholder={placeholder}
        onChange={mockOnChange}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders input element with correct attributes', () => {
    render(
      <Input
        name={name}
        label={label}
        type={type}
        placeholder={placeholder}
        onChange={mockOnChange}
      />,
    );

    const inputElement = screen.getByPlaceholderText(placeholder);

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('name', name);
    expect(inputElement).toHaveAttribute('type', type);
  });
});
