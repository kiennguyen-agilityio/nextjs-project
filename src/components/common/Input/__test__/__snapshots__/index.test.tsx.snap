// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-dark-100 text-sm font-medium leading-5 mb-4"
      for="testName"
    >
      Test Label
    </label>
    <input
      class="rounded-[10px] w-full h-12 py-2 px-4 bg-white-300 text-dark-90 placeholder-dark-50 leading-tight border border-gray-300 focus:outline-none focus:shadow-outline focus:ring-2 focus:ring-slate-300 "
      id="testName"
      name="testName"
      placeholder="Test Placeholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
